### PR TITLE
feat(RayTracing): add helpers to check a point is inside a closed sur…

### PR DESCRIPTION
**Pull request by BenPinet**
…face

### CODE REVIEW

File: src/geode/model/helpers/ray_tracing.cpp [hunk 1]
```diff
@@ -102,6 +90,47 @@ namespace geode
...
+            geode::Vector3D{ { 1., 0., 0.3 } },
+            geode::Vector3D{ { 1., 0., 0.5 } },
+            geode::Vector3D{ { 0., 1., 0. } },
+            geode::Vector3D{ { 0.1, 1., 0. } },
+            geode::Vector3D{ { 0.3, 1., 0. } },
+            geode::Vector3D{ { 0.5, 1., 0. } },
+            geode::Vector3D{ { 0., 0., 1. } },
+            geode::Vector3D{ { 0., 0.1, 1. } },
+            geode::Vector3D{ { 0., 0.3, 1. } },
+            geode::Vector3D{ { 0., 0.5, 1. } } } };
```
Comment (panquez) [inline src/geode/model/helpers/ray_tracing.cpp:106]: Avoid duplication of these Vectors.

File: src/geode/model/helpers/ray_tracing.cpp [hunk 2]
```diff
@@ -102,6 +90,47 @@ namespace geode
...
+        std::vector< std::reference_wrapper< const geode::SurfaceMesh3D > >
+            border_to_merge;
+        for( const auto& surface : brep.boundaries( block ) )
+        {
+            border_to_merge.emplace_back( surface.mesh() );
+        }
+        geode::detail::SurfaceMeshMerger3D border_merger{ border_to_merge,
```
Comment (panquez) [inline src/geode/model/helpers/ray_tracing.cpp:113]: Do you really need to merge surfaces to count number of intersections?

File: src/geode/model/helpers/ray_tracing.cpp [hunk 3]
```diff
@@ -34,47 +34,44 @@
 
 namespace
 {
-    geode::BoundarySurfaceIntersections find_intersections_with_boundaries(
-        const geode::Ray3D& ray,
-        const geode::BRep& brep,
-        const geode::Block3D& block )
+    static std::array< geode::Vector3D, 12 > directions = { { geode::Vector3D{
```
[the remark this answers is not in the document]
Comment (BotellaA) [reply src/geode/model/helpers/ray_tracing.cpp:37]: Je confirme, c'est important, idéalement il faudrait pouvoir le mettre constexpr mais je ne sais pas si c'est possible

**Review by panquez — CHANGES_REQUESTED** [review]

**Pull request merged**